src/components/OptionListContextProvider.tsx

```diff
@@ -64,22 +64,11 @@ function OptionsListContextProvider({reports, children}: OptionsListProviderProp
             return;
         }
 
-        const lastUpdatedReport = ReportUtils.getLastUpdatedReport();
-
-        if (!lastUpdatedReport) {
-            return;
-        }
-
-        const newOption = OptionsListUtils.createOptionFromReport(lastUpdatedReport, personalDetails);
-        const replaceIndex = options.reports.findIndex((option) => option.reportID === lastUpdatedReport.reportID);
-
-        if (replaceIndex === -1) {
-            return;
-        }
+        const newReports = OptionsListUtils.createOptionList({}, reports).reports;
 
         setOptions((prevOptions) => {
             const newOptions = {...prevOptions};
-            newOptions.reports[replaceIndex] = newOption;
```

Comment:
Just wanted to note that the old code was mutating data here which is pretty bad in React.